torch/csrc/jit/fusers/cpu/fusion_compiler.cpp

```diff
@@ -14,19 +14,95 @@
 #include <sstream>
 #include <tuple>
 
+#if ((__GLIBC__ == 2) && (__GLIBC_MINOR__ >= 2)) || (__GLIBC__ > 2)
+#include <spawn.h>
+#ifdef POSIX_SPAWN_USEVFORK
+#define USE_POSIX_SPAWN
+#include <sys/wait.h>
+extern char **environ;
+#endif
+#endif
+
+#ifndef USE_POSIX_SPAWN
+#ifndef __APPLE__
+#include <malloc.h>
+#define USE_MALLOC_TRIM
+#endif
+#endif
+
 namespace torch { namespace jit { namespace cpufuser {
 
 CPUFusionCompiler& getFusionCompiler() {
   static CPUFusionCompiler compiler;
   return compiler;
 }
 
+#ifdef USE_POSIX_SPAWN
+struct PosixSpawnAttrVfork {
+  explicit PosixSpawnAttrVfork() {
+    int err = posix_spawnattr_init(&attr);
+    if (err != 0) {
+      AT_ERROR("posix_spawnattr_init: ", strerror(err));
+    }
+
+    err = posix_spawnattr_setflags(&attr, POSIX_SPAWN_USEVFORK);
+    if (err != 0) {
+      AT_ERROR("posix_spawnattr_setflags: ", strerror(err));
+    }
+  }
+  ~PosixSpawnAttrVfork() {
+    int err = posix_spawnattr_destroy(&attr);
+    if (err != 0) {
+      AT_WARN("posix_spawnattr_destroy: ", strerror(err));
+    }
+  }
+  posix_spawnattr_t attr;
+};
+#endif // USE_POSIX_SPAWN
+
+int runCommand(const std::string& command) {
+#ifdef USE_POSIX_SPAWN
+  // NB: Even with copy-on-write, fork can fail if the parent process's resident
+  // memory usage is more than half the total RAM. Because of this, we avoid
+  // system() (that calls fork) and try to use posix_spawn whenever possible.
+  PosixSpawnAttrVfork attr;
+  pid_t pid;
+
+  // Run sh to run the provided command
+  const char* cmd = command.c_str();
+  std::vector<char> cmd_copy(cmd, cmd + command.size() + 1);
+  char* argv[] = {"sh", "-c", cmd_copy.data(), NULL};
```

Comment:
Does this really work? Don't you have to supply every single argument separately? Also, won't the compiler complain about `char*[]` instead of `const char*[]`? I remember we had a bunch of warnings for that, because that's what Python often uses.

Reply:
as far as I can tell it works because it's executing a `sh -c "your command here"` where "your command here" is the command we want and its arguments.

The compiler hasn't complained to me yet, I'm not sure why...